ffi/src/expressions/engine.rs

```diff
@@ -11,9 +11,14 @@ use delta_kernel::expressions::{
 };
 use delta_kernel::DeltaResult;
 
+pub enum ExpressionOrPredicate {
```

Comment:
can this be `pub(crate)`? I only see it being used within `KernelExpressionVisitorState` (privately)?